src/include/cpu/rpp_cpu_common.hpp

```diff
@@ -3135,9 +3135,20 @@ inline void compute_color_temperature_24_host(__m256 *p, __m256 pAdj)
     p[2] = _mm256_sub_ps(p[2], pAdj);    // color_temperature adjustment Bs
 }
 
-inline void compute_fog_48_host(__m256 *p, __m256 *pFogAlphaMask, __m256 *pFogIntensityMask, __m256 pIntensityFactor)
-{
-    __m256 pAlphaMaskFactor[2], pIntensityMaskFactor[2];
+inline void compute_fog_48_host(__m256 *p, __m256 *pFogAlphaMask, __m256 *pFogIntensityMask, __m256 pIntensityFactor, __m256 pGrayFactor)
+{
+    __m256 pAlphaMaskFactor[2], pIntensityMaskFactor[2], pGray[2], pOneMinusGrayFactor;
+    pGray[0] = _mm256_add_ps(_mm256_mul_ps(_mm256_set1_ps(0.299f), p[0]), _mm256_add_ps(_mm256_mul_ps(_mm256_set1_ps(0.587f), p[2]), _mm256_mul_ps(_mm256_set1_ps(0.114f), p[4]))); 
+    pGray[1] = _mm256_add_ps(_mm256_mul_ps(_mm256_set1_ps(0.299f), p[1]), _mm256_add_ps(_mm256_mul_ps(_mm256_set1_ps(0.587f), p[3]), _mm256_mul_ps(_mm256_set1_ps(0.114f), p[5])));
```

Comment:
Couple of changes
1. Please store this 3 __m256 const values in an array in fog kernel here, instead of doing set1_ps operation each time
https://github.com/r-abishek/rpp/blob/sr/fog_pr_changes/src/modules/cpu/kernel/fog.hpp#L84

```
__m256 pConversionFactor[3];
 pConversionFactor[0] = _mm256_set1_ps(0.299f);
 pConversionFactor[1] = _mm256_set1_ps(0.587f);
 pConversionFactor[2] = _mm256_set1_ps(0.114f);
```
Pass this array to compute_fog_48_host and use it

2. Use fmadd operation for L3141 and L3142 since we are doing multiplication followed by addition

Reply:
is this done?